src/middlewared/middlewared/plugins/iscsi.py

```diff
@@ -833,34 +833,22 @@ async def disk_choices(self):
         """
         diskchoices = {}
 
-        zvol_query_filters = [('type', '=', 'DISK')]
-
-        used_zvols = [
-            i['path'] for i in (await self.query(zvol_query_filters))
-        ]
-
         zvols = await self.middleware.call(
-            'pool.dataset.query', [
-                ('type', '=', 'VOLUME'),
-                ('locked', '=', False)
-            ]
-        )
-        zvol_list = []
-        for zvol in zvols:
-            zvol_name = zvol['name']
-            zvol_size = zvol['volsize']['value']
-            zvol_list.append(zvol_name)
-            key = os.path.relpath(zvol_name_to_path(zvol_name), '/dev')
-            if key not in used_zvols:
-                diskchoices[key] = f'{zvol_name} ({zvol_size})'
-
-        zfs_snaps = await self.middleware.call(
```

Comment:
We are not getting this anymore i believe ?

Reply:
```
root@truenas[~]# midclt call zfs.dataset.available_zvols_fast | jq
{
  "dozer/TESTVOL@now": {
    "name": "dozer/TESTVOL@now",
    "dev": "zd32"
  },
  "dozer/TESTVOL": {
    "name": "dozer/TESTVOL",
    "dev": "zd0"
  },
  "dozer/smb-vss/sub1/zvol02": {
    "name": "dozer/smb-vss/sub1/zvol02",
    "dev": "zd16"
  }
```

They are returned _if_ the ZFS snapdev is visible. I'm not sure how this was supposed to work with a hidden snapdev on Linux (probably doesn't since it's not available through /sys/block).

Reply:
Created NAS-116482 for someone else to investigate interaction between snapdev property and exposing snapshots of zvols.